haystack/nodes/prompt/prompt_node.py

```diff
@@ -864,25 +865,23 @@ def run(
         prompt template.
         :param documents: The documents to be used for the prompt.
         :param meta: The meta to be used for the prompt. Usually not used.
+        :param invocation_context: The invocation context to be used for the prompt.
         """
 
-        if not meta:
-            meta = {}
         # invocation_context is a dictionary that is passed from a pipeline node to a pipeline node and can be used
         # to pass results from a pipeline node to any other downstream pipeline node.
-        if "invocation_context" not in meta:
-            meta["invocation_context"] = {}
+        invocation_context = invocation_context or {}
 
         results = self(
             query=query,
             labels=labels,
             documents=[doc.content for doc in documents if isinstance(doc.content, str)] if documents else [],
-            **meta["invocation_context"],
+            **invocation_context,
         )
```

Comment:
Due to the comment above, this part would need to change too:

```suggestion
        for doc in invocation_context.get("documents", []):
            if not isinstance(doc, str) and not isinstance(doc.content, str):
                raise ValueError("PromptNode only accepts text documents.")
        invocation_context["documents"] = [
            doc.content if isinstance(doc, Document) else doc for doc in invocation_context.get("documents", [])
        ]
                
        results = self(**invocation_context)
```

Reply:
I don't get why we are merging documents with the invocation context here? I think I am missing something here.

Reply:
Ok I get it now.